srml/democracy/src/lib.rs

```diff
@@ -292,51 +335,69 @@ impl<T: Trait> Module<T> {
 		Self::deposit_event(RawEvent::Executed(index, ok));
 	}
 
+	fn launch_next(now: T::BlockNumber) -> Result {
+		let mut public_props = Self::public_props();
+		if let Some((winner_index, _)) = public_props.iter()
+			.enumerate()
+			.max_by_key(|x| Self::locked_for((x.1).0).unwrap_or_else(Zero::zero)/*defensive only: All current public proposals have an amount locked*/)
+		{
+			let (prop_index, proposal, _) = public_props.swap_remove(winner_index);
+			<PublicProps<T>>::put(public_props);
+
+			if let Some((deposit, depositors)) = <DepositOf<T>>::take(prop_index) {//: (T::Balance, Vec<T::AccountId>) =
+				// refund depositors
+				for d in &depositors {
+					<balances::Module<T>>::unreserve(d, deposit);
+				}
+				Self::deposit_event(RawEvent::Tabled(prop_index, deposit, depositors));
+				Self::inject_referendum(now + Self::voting_period(), proposal, VoteThreshold::SuperMajorityApprove, Self::public_delay())?;
+			}
+		}
+
+		Ok(())
+	}
+
+	fn bake_referendum(now: T::BlockNumber, index: ReferendumIndex, info: ReferendumInfo<T::BlockNumber, T::Proposal>) -> Result {
+		let (approve, against, capital) = Self::tally(index);
+		let total_issuance = <balances::Module<T>>::total_issuance();
+		let approved = info.threshold.approved(approve, against, capital, total_issuance);
+		let lock_period = Self::public_delay();
+
+		for (a, vote) in Self::voters_for(index).into_iter()
+			.map(|a| (a.clone(), Self::vote_of((index, a))))
+			// ^^^ defensive only: all items come from `voters`; for an item to be in `voters` there must be a vote registered; qed
+			.filter(|&(_, vote)| vote.is_aye() == approved)
+		{
+			let locked_until = now + lock_period * T::BlockNumber::sa((vote.multiplier()) as u64);
```

Comment:
Shouldn't the multiplier only be used if the referendum was approved?

Reply:
(answering this in the documentation of the module or function would be ideal)

Reply:
If it was rejected only those who voted against get their bond locked. Maybe this merits a comment in the `filter` above.

Reply:
The logic is correct; I added a few comments and a link to my presentation.